### ISSUE TITLE
Document remains in edit/lock mode, even document is already stored and closed

### ISSUE THREAD

**Issue by florianrhomberg** [ONLYOFFICE/onlyoffice-alfresco#182]
Hi,
I have a test environment of Alfresco 7 using Bitnami and an OnlyOffice Server Community on a different machine. However when I open in Alfresco a document to edit, make changes store and then close the tab, the document still remains in edit mode, even after hours. I have to cancel and therefore unlock the document manually in Alfresco.

Is this a bug, or do I have a misconfiguration?

Thanks,
Florian

**Comment by AFaust** [comment]
Not sure if you have the same error that I currently have. Looking at the logs and then source code, I see a programmer mistake in a config lookup. Specifically, if I don't set the `onlyoffice.cert=true|false` property, I get a ClassCastException when OnlyOffice tries to save back the state of the edited document. OnlyOffice will still say "All changes were saved" despite an HTTP 500 error in the background.
The following is the error as shown in OnlyOffice logs:

```
[2023-11-05T16:19:14.911] [ERROR] [localhost] [e1ad77fc-5712-4178-81f2-e99977ea6a71_1.0] [admin1] nodeJS - sendServerRequest error: url = http://repository:8080/alfresco/s/parashift/onlyoffice/callback?nodeRef=workspace://SpacesStore/e1ad77fc-5712-4178-81f2-e99977ea6a71&cb_key=dV9WHjHtg3QJ5uVBXNog0FMHvtx5Zm6Da7UR6OMyq_4;data = {"key":"e1ad77fc-5712-4178-81f2-e99977ea6a71_1.0","status":2,"url":"http://localhost:8080/cache/files/data/e1ad77fc-5712-4178-81f2-e99977ea6a71_1.0_3379/output.pptx/output.pptx?md5=bSFAnQ81KEh2fGyGMsOGzQ&expires=1699202055&filename=output.pptx","history":{},"users":["admin"],"actions":[{"type":0,"userid":"admin"}],"lastsave":"2023-11-05T15:59:57.000Z","notmodified":false,"token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJrZXkiOiJlMWFkNzdmYy01NzEyLTQxNzgtODFmMi1lOTk5NzdlYTZhNzFfMS4wIiwic3RhdHVzIjoyLCJ1cmwiOiJodHRwOi8vbG9jYWxob3N0OjgwODAvY2FjaGUvZmlsZXMvZGF0YS9lMWFkNzdmYy01NzEyLTQxNzgtODFmMi1lOTk5NzdlYTZhNzFfMS4wXzMzNzkvb3V0cHV0LnBwdHgvb3V0cHV0LnBwdHg_bWQ1PWJTRkFuUTgxS0VoMmZHeUdNc09HelEmZXhwaXJlcz0xNjk5MjAyMDU1JmZpbGVuYW1lPW91dHB1dC5wcHR4IiwiaGlzdG9yeSI6e30sInVzZXJzIjpbImFkbWluIl0sImFjdGlvbnMiOlt7InR5cGUiOjAsInVzZXJpZCI6ImFkbWluIn1dLCJsYXN0c2F2ZSI6IjIwMjMtMTEtMDVUMTU6NTk6NTcuMDAwWiIsIm5vdG1vZGlmaWVkIjpmYWxzZSwiZmlsZXR5cGUiOiJwcHR4IiwiaWF0IjoxNjk5MjAxMTU0LCJleHAiOjE2OTkyMDE0NTR9.O7HEq9DTJ4NVIG8WTRNMAka-T0ijuCnFSmiYW8Rv_Dw","filetype":"pptx"} Error: Error response: statusCode:500; headers:{"x-frame-options":"SAMEORIGIN","content-length":"176","date":"Sun, 05 Nov 2023 16:19:14 GMT","connection":"close","server":""}; body: {"error":1, "message":"class java.lang.Boolean cannot be cast to class java.lang.String (java.lang.Boolean and java.lang.String are in module java.base of loader 'bootstrap')"}
```

The ACS log shows the following error:

```
2023-11-05T16:19:14,901 [] ERROR [onlyoffice.scripts.CallBack] [http-nio-8080-exec-9] Error execution script Callback
java.lang.ClassCastException: class java.lang.Boolean cannot be cast to class java.lang.String (java.lang.Boolean and java.lang.String are in module java.base of loader 'bootstrap')
at com.parashift.onlyoffice.util.ConfigManager.getAsBoolean(ConfigManager.java:68) ~[onlyoffice-integration-platform-jar-6.1.0.jar:?]
at com.parashift.onlyoffice.util.RequestManager.getHttpClient(RequestManager.java:120) ~[onlyoffice-integration-platform-jar-6.1.0.jar:?]
at com.parashift.onlyoffice.util.RequestManager.executeRequest(RequestManager.java:90) ~[onlyoffice-integration-platform-jar-6.1.0.jar:?]
at com.parashift.onlyoffice.util.RequestManager.executeRequestToDocumentServer(RequestManager.java:55) ~[onlyoffice-integration-platform-jar-6.1.0.jar:?]
...
```

**Comment by nhducseuit** [comment]
I have the same question

**Comment by aleksandrfedorov97** [comment]
fixed in #194
